### PR TITLE
EXAPP-18 Portal and Listing Field Mapping.

### PR DESCRIPTION
**Pull request by Karan3036**
Created three LWC Component and One Apex class for Portal and Listing Object field mapping.

### CODE REVIEW

File: force-app/main/default/lwc/newPopUp/newPopUp.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { LightningElement, api, track } from 'lwc';
+import savePropertyPortalRecord from '@salesforce/apex/controlCenterController.savePropertyPortalRecord';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+export default class NewPopUp extends LightningElement {
+
```
Comment (MRG-MV) [inline force-app/main/default/lwc/newPopUp/newPopUp.js:5]: Add a decorator for all the variables and for the methods follow the sequence 
method sequence:- wire method, Connected callback, rendered callback, event handler method, other js methods

Also Add following comments for the all the methods
/**
* Method Name: previewFormcmp
* @description: description for method with use
* Date: 03/06/2024
* Created By: Developer name
*/

File: force-app/main/default/lwc/newPopUp/newPopUp.css [hunk 2]
```diff
@@ -0,0 +1,55 @@
+.input-container {
```
Comment (MRG-MV) [inline force-app/main/default/lwc/newPopUp/newPopUp.css:1]: try to remove CSS given in 'px' try using '%' or 'rem'

File: force-app/main/default/lwc/portalMappingComponent/portalMappingComponent.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    handlehidepopup(event) {
+        this.showModal = event.details;
+    }
+
+    handlehideandrefreshpage(event) {
+        this.showModal = event.details;
+        this.getPortalRecord()
+    }
+
```
Comment (MRG-MV) [inline force-app/main/default/lwc/portalMappingComponent/portalMappingComponent.js:45]: define variables at the start also add decorators for all the variables

File: force-app/main/default/lwc/portalMappingComponent/portalMappingComponent.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                } else {
+                    this.isPortalData = false;
+                }
+            })
+            .catch(error => {
+                console.error('Error fetching Listing field data', error);
+            });
+    }
+
+    handlehidepopup(event) {
```
Comment (MRG-MV) [inline force-app/main/default/lwc/portalMappingComponent/portalMappingComponent.js:37]: use camelcase for method name

File: force-app/main/default/lwc/portalMappingLandingPage/portalMappingLandingPage.js [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+export default class PortalMappingLandingPage extends NavigationMixin(LightningElement) {
+
+    isInitalRender = true;
+    @api portalId;
+    @api portalGen;
+    @api portalName;
+    @api portalIconUrl;
+    @api portalStatus;
+
+    showModal;
```
Comment (MRG-MV) [inline force-app/main/default/lwc/portalMappingLandingPage/portalMappingLandingPage.js:17]: assign default value and decorator.

File: force-app/main/default/lwc/portalMappingLandingPage/portalMappingLandingPage.js [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+
+        this.originalMappingData = this.finalList;
+    }
+
+    getListingLabel(listingFieldValue) {
+        const listingOption = this.MainListingOptions.find(option => option.apiName === listingFieldValue);
+        return listingOption ? listingOption.label : '';
+    }
+
+    renderedCallback() {
```
Comment (MRG-MV) [inline force-app/main/default/lwc/portalMappingLandingPage/portalMappingLandingPage.js:97]: follow the order of methods and add comments for each methods

**Review by MRG-MV — CHANGES_REQUESTED** [review]
make all the requested changes and follow the shared document for further reference

File: force-app/main/default/classes/PortalMappingController.cls [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+                    return listOfPortalRecords;
+                } else {
+                    return null;
+                }
+        } catch (Exception e) {
+            throw new AuraHandledException(e.getMessage());
+        }
+    }
+
+    public class FieldWrapper {
```
[the remark this answers is not in the document]
Comment (Karan3036) [reply force-app/main/default/classes/PortalMappingController.cls:35]: I have added the description for wrapper class as well.

File: force-app/main/default/lwc/newPopUp/newPopUp.css (hunk 2 above)
Comment (Karan3036) [reply force-app/main/default/lwc/newPopUp/newPopUp.css:1]: I have replaced the px with rem and per. Only font-size value is in px.

File: force-app/main/default/lwc/portalMappingComponent/portalMappingComponent.js (hunk 4 above)
Comment (Karan3036) [reply force-app/main/default/lwc/portalMappingComponent/portalMappingComponent.js:37]: I have changed the method name.

File: force-app/main/default/lwc/portalMappingComponent/portalMappingComponent.js (hunk 3 above)
Comment (Karan3036) [reply force-app/main/default/lwc/portalMappingComponent/portalMappingComponent.js:45]: I have moved the variables at start and also added the decorators as well.

File: force-app/main/default/lwc/portalMappingLandingPage/portalMappingLandingPage.js (hunk 6 above)
Comment (Karan3036) [reply force-app/main/default/lwc/portalMappingLandingPage/portalMappingLandingPage.js:97]: Arranged the methods in proper order.

File: force-app/main/default/lwc/portalMappingLandingPage/portalMappingLandingPage.js (hunk 5 above)
Comment (Karan3036) [reply force-app/main/default/lwc/portalMappingLandingPage/portalMappingLandingPage.js:17]: Added the decorators and assigned default values to the variables.

**Review by MRG-MV — APPROVED** [review]

**Pull request merged**